riscv/triggers.cc

```diff
@@ -273,9 +273,9 @@ void mcontrol6_t::tdata1_write(processor_t * const proc, const reg_t val, const
   dmode = get_field(val, CSR_MCONTROL6_DMODE(xlen));
   vs = get_field(val, CSR_MCONTROL6_VS);
   vu = get_field(val, CSR_MCONTROL6_VU);
-  hit = get_field(val, CSR_MCONTROL6_HIT);
+  hit = hit_t(2 * get_field(val, CSR_MCONTROL6_HIT1) + get_field(val, CSR_MCONTROL6_HIT0)); // 2-bit field {hit1,hit0}
   select = get_field(val, CSR_MCONTROL6_SELECT);
-  timing = legalize_timing(val, CSR_MCONTROL6_TIMING, CSR_MCONTROL6_SELECT, CSR_MCONTROL6_EXECUTE, CSR_MCONTROL6_LOAD);
+  timing = legalize_timing(val, 0, CSR_MCONTROL6_SELECT, CSR_MCONTROL6_EXECUTE, CSR_MCONTROL6_LOAD);
```

Comment:
I think this means:
* data load triggers trigger right after the event
* all other triggers trigger right before the event

That means you can't combine a data load trigger with an address trigger to trigger on a load of a value at a given address. gdb doesn't support setting triggers in that way so possibly we don't care. I think it's worth clarifying the behavior with a comment, since it's not super obvious just from reading this code.

Reply:
Add a comment in https://github.com/riscv-software-src/riscv-isa-sim/pull/1257/commits/33e992d5e4bd23cac673330efcf833e6bd3918f3 to clarify the behavior.